dev/TabView/InteractionTests/TabViewTests.cs

```diff
@@ -140,6 +140,21 @@ public void TabSizeAndScrollButtonsTest()
                 // With largerTab now rendering wider, the scroll buttons should appear:
                 Verify.IsTrue(AreScrollButtonsVisible(), "Scroll buttons should appear");
 
+                // Scroll all the way to the left and verify decrease/increase button visual state
+                FindElement.ByName<Button>("ScrollTabViewToTheLeft").InvokeAndWait();
+                Verify.IsFalse(IsScrollDecreaseButtonEnabled(), "Scroll decrease button should be disabled");
+                Verify.IsTrue(IsScrollIncreaseButtonEnabled(), "Scroll increase button should be enabled");
+
+                // Scroll to the middle position and verify decrease/increase button visual state
+                FindElement.ByName<Button>("ScrollTabViewToTheMiddle").InvokeAndWait();
+                Verify.IsTrue(IsScrollDecreaseButtonEnabled(), "Scroll decrease button should be enabled");
+                Verify.IsTrue(IsScrollIncreaseButtonEnabled(), "Scroll increase button should be enabled");
+
+                // Scroll all the way to the right and verify decrease/increase button visual state
+                FindElement.ByName<Button>("ScrollTabViewToTheRight").InvokeAndWait();
+                Verify.IsTrue(IsScrollDecreaseButtonEnabled(), "Scroll decrease button should be enabled");
+                Verify.IsFalse(IsScrollIncreaseButtonEnabled(), "Scroll increase button should be disabled");
```

Comment:
This assert fails on my machine 100% of the time, however adding a small delay after line 153 (e.g. 300ms) seems to fix that issue. I am not sure if it's just my machine or if this is generally an issue. Did the test run perfectly on your machine @JasonStein ?

Reply:
Yes, it runs fine on my machine but I have only tested once. Let me check again.

Reply:
You are right, it does not work if there is no delay (not sure why it works on my machine previously). Because the view state change triggered after scroll viewer view changed event so it is not going to happen right away. I added a delay after each invoke. Please check latest iteration.

Reply:
Test looks fine now.

Reply:
Can we do Wait.ForIdle instead ? Thread.Sleep /Wait for milliseconds should be avoided in general, they tend to make the test slower on fast machines and fail on really slow machines causing the test to be flaky. Wait.ForIdle is designed for this exact purpose - once the system is done with everything it will come out of the wait and that time will be less on faster machines.

Reply:
Calling InvokeAndWait also calls WaitForIdle down the line, however the test did not work without waiting.

Reply:
@kmahone @llongley Any ideas why WaitForIdle might not be working here ?

Reply:
I tried WaitForIdle and it is actually working on my machine. @chingucoding Can you confirm?

Reply:
@ranjeshj Can you help trigger a remote build to see if it works on test machine?

Reply:
I've tried your code and it still fails. Here a recording of the test run. Notice that the app finishes the test way before we have scrolled all the way to the right:

![tabview-test](https://user-images.githubusercontent.com/16122379/77688020-abbd1400-6f9f-11ea-9d63-fe6ca460359c.gif)

Reply:
Interesting, btw, I used to found rendering issue with ScrollViewer's ChangeView in my app as well. Although I set animation to be false, it still takes a good amount of time before taking effect. And if we do it before loading, it might not work at all.

Maybe let's revert it back to use Wait.For for now?

Reply:
It seems that `Wait.ForMilliseconds` is the only reliable solution so far, so it might be the best to use that for now. :/

Reply:
@ranjeshj Do you agree?

Reply:
Another option could be to in the app, look for the ScrollViewer in the tree, hookup to its ViewChanged and in the handler set a toggle button or checkbox or similar. Then from the test you can query/wait on that from UIA. We have had to deal with time based delays in the past and it has caused trouble, so this bit of extra work is worth it.

Reply:
@chingucoding I tried to get the scroll viewer out of TabView in test UI page but I do not see a way of doing it using the existing VisualTreeHelper. Can you take a look into it? Otherwise I will be blocked on this. Thanks!

Reply:
Let me try that.

Reply:
> @chingucoding I tried to get the scroll viewer out of TabView in test UI page but I do not see a way of doing it using the existing VisualTreeHelper. Can you take a look into it? Otherwise I will be blocked on this. Thanks!
@JasonStein Do you still want me to look into it? 😅

Reply:
No I am fine. Let me try first. Is there any existing example on how to wait on a particular text box to be updated?  "Then from the test you can query/wait on that from UIA."

Reply:
There is some event in the WaitHelper.cs file which technically should work, though the fact that this exact class is letting us down here is not to promising. The easies/"safest" way would be to wait for a certain amount of time and retry. This is also something the WaitHelper does in some cases (I think).

Reply:
Even with wait on ViewChanged, I am still smelling race condition here since "IsEnabled" is set after ViewChanged not as part of ViewChanged event. It is unlikely but in theory, it could happen. @ranjeshj Should we keep 300ms waiting here for now?

Reply:
There are some unreliable tests, however we try to keep avoiding unreliable tests mostly. If waiting 300ms or 500ms or 1000ms makes it stable, that would be a possible solution, though it may create problems in the future.

Reply:
@JasonStein The tests are PR gates, so if tests are flaky it will start blocking future PRs. We should try as much as possible to avoid that. Please file a test issue to investigate this and to unblock this PR for now. If the test starts failing, then we can disable it and link it to the issue. Please also add a comment next to the delay pointing to the issue.

Reply:
I tried to implement ViewChanged and even with that, it is not going to work. Then I realized that I thought I disabled scrolling animation, in fact, I did not. Now I have disabled all scrolling animation and things started to work and test is now passing with Wait.ForIdle.